File: leetcode-solutions/0134-gas-station/solution.java
class Solution {
    public int canCompleteCircuit(int[] g,int[] c){
        int tot=0,tank=0,start=0;
        for(int i=0;i<g.length;i++){
            int gain = g[i]-c[i];
            tot+=gain;
            tank+=gain;
            if(tank<0){
                start=i+1;
                tank=0;
            }

        }
    if(tot<0)
    return -1;
    return start;
    }
    }

File: leetcode-solutions/0134-gas-station/solution.py
class Solution(object):
    def canCompleteCircuit(self, gas, cost):


        """
        :type gas: List[int]
        :type cost: List[int]
        :rtype: int
        """
        tot=start=tank=0
        for i in range(len(gas)):
            
            gain=gas[i]-cost[i]
            tot+=gain
            tank+=gain
            if tank<0:
                start=i+1
                tank=0
        if tot<0:
            return -1
        return start
        

